chromadb/cli/cli.py

```diff
@@ -83,6 +93,88 @@ def run(
     uvicorn.run(**config)
 
 
+@utils_app.command()  # type: ignore
+def vacuum(
+    path: str = typer.Option(
+        help="The path to a Chroma data directory.",
+    ),
+    force: bool = typer.Option(False, help="Force vacuuming without confirmation."),
+) -> None:
+    """
+    Vacuum the database. This may result in a small increase in performance.
+
+    If you recently upgraded Chroma from a version below 0.6 to 0.6 or above, you should run this command once to greatly reduce the size of your database and enable continuous database pruning. In most other cases, vacuuming will save very little disk space.
+
+    The execution time of this command scales with the size of your database. It block both reads and writes to the database while it is running.
+    """
+    console = Console(
+        highlight=False
+    )  # by default, rich highlights numbers which makes the output look weird when we try to color numbers ourselves
+
+    if not os.path.exists(path):
+        console.print(f"[bold red]Path {path} does not exist.[/bold red]")
+        raise typer.Exit(code=1)
+
+    if not os.path.exists(f"{path}/chroma.sqlite3"):
+        console.print(
+            f"[bold red]Path {path} is not a Chroma data directory.[/bold red]"
+        )
+        raise typer.Exit(code=1)
+
+    if not force and not typer.confirm(
+        "Are you sure you want to vacuum the database? This will block both reads and writes to the database and may take a while."
```

Comment:
recommend not to run while server is running

Reply:
Do we / can we emit some sort of progress information, and also inform the user when it's done?

Reply:
please see the recording in OP
unfortunately we don't have any kind of granular progress / estimated remaining time available, but we do display a spinner